x-pack/plugins/ingest_manager/common/types/models/datasource.ts

```diff
@@ -5,31 +5,23 @@
  */
 
 export interface NewDatasource {
-  name: string;
-  namespace?: string;
-  config_id: string;
-  enabled: boolean;
+  enabled?: boolean;
+  title?: string;
   package?: {
-    assets: Array<{
-      id: string;
-      type: string;
-    }>;
-    description: string;
     name: string;
-    title: string;
     version: string;
   };
-  output_id: string;
+  namespace?: string;
+  use_output: string;
   inputs: Array<{
     type: string;
-    enabled: boolean;
     processors?: string[];
     streams: Array<{
-      id: string;
-      enabled: boolean;
-      dataset: string;
-      processors?: string[];
-      config?: Record<string, any>;
+      id?: string;
+      enabled?: boolean;
+      dataset?: string;
+      metricset?: string;
+      paths?: string[];
```

Comment:
Question: don't we need something like `config` (from prior version) that accepts free-typed JSON? 

I'm thinking of the use case for Endpoint where we will need to add our policy configuration data.
The example in the beats repo has two suggestions for this use case (line [472](https://github.com/elastic/beats/blob/3bb05459201ac61090b0fac4c66aa050a94f19f9/x-pack/agent/docs/agent_configuration_example.yml#L472) and [578](https://github.com/elastic/beats/blob/3bb05459201ac61090b0fac4c66aa050a94f19f9/x-pack/agent/docs/agent_configuration_example.yml#L528) - I thought that from one of the review meetings, the Endpoint team (@ferullo ) indicated suggestion 2 was best but @ph  can confirm.